### PR TITLE
Fix `CodeGenerationFormat` to correctly handle trailing trivia when formatting list elements

### PR DESCRIPTION
**Pull request by roopekv**
This fixes 2 issues with how `CodeGenerationFormat` handles trailing trivia when formatting lists to be separated by a newline. These issues don't manifest in the current codebase, but the fixes in this PR will prevent them from appearing in the future. Examples are provided below.

### Example

```swift
DeclSyntax(
  """
  public static func \(tokenSpec.funcDeclName)(
    _ value: Keyword, // EXAMPLE <--
    leadingTrivia: Trivia = [],
    trailingTrivia: Trivia = [],
    presence: SourcePresence = .present
  ) -> TokenSyntax {
    // ...
  }
  """
)
```
#### Before (missing space between `,` and `//`)
```swift
public static func keyword(
  _ value: Keyword,// EXAMPLE <--
  leadingTrivia: Trivia = [],
  trailingTrivia: Trivia = [],
  presence: SourcePresence = .present
) -> TokenSyntax {
  // ...
}
```
#### After
```swift
public static func keyword(
  _ value: Keyword, // EXAMPLE <--
  leadingTrivia: Trivia = [],
  trailingTrivia: Trivia = [],
  presence: SourcePresence = .present
) -> TokenSyntax {
  // ...
}
```

### Another example
```swift
DeclSyntax(
  """
  func verify<Node: RawSyntaxNodeProtocol>(_ raw: RawSyntax?, /* EXAMPLE <-- */ as _: Node.Type, file: StaticString = #fileID, line: UInt = #line) -> ValidationError? {
    // ...
  }
  """
)
```
#### Before (missing space between `,` and `/*`)
```swift
func verify<Node: RawSyntaxNodeProtocol>(
  _ raw: RawSyntax?,/* EXAMPLE <-- */ 
  as _: Node.Type,
  file: StaticString = #fileID,
  line: UInt = #line
) -> ValidationError? {
  // ...
}
```
#### After
```swift
func verify<Node: RawSyntaxNodeProtocol>(
  _ raw: RawSyntax?, /* EXAMPLE <-- */ 
  as _: Node.Type,
  file: StaticString = #fileID,
  line: UInt = #line
) -> ValidationError? {
  // ...
}
```

### Example

```swift
DeclSyntax(
  """
  func verify<Node: RawSyntaxNodeProtocol>(_ raw: RawSyntax?, as _: Node.Type, file: StaticString = #fileID, line: UInt = #line /* EXAMPLE <-- */) -> ValidationError? {
    // ...
  }
  """
)
```

#### Before (missing  `/* EXAMPLE <-- */`)
```swift
func verify<Node: RawSyntaxNodeProtocol>(
  _ raw: RawSyntax?,
  as _: Node.Type,
  file: StaticString = #fileID,
  line: UInt = #line
) -> ValidationError? {
  // ...
}
```

#### After
```swift
func verify<Node: RawSyntaxNodeProtocol>(
  _ raw: RawSyntax?,
  as _: Node.Type,
  file: StaticString = #fileID,
  line: UInt = #line /* EXAMPLE <-- */
) -> ValidationError? {
  // ...
}
```

### CODE REVIEW

File: CodeGeneration/Sources/Utils/CodeGenerationFormat.swift [hunk 1]
```diff
@@ -158,21 +158,30 @@ public class CodeGenerationFormat: BasicFormat {
     }
     formattedChildren = formattedChildren.map { child in
       var child = child
-      child.trailingTrivia = Trivia(pieces: child.trailingTrivia.drop(while: \.isSpaceOrTab))
+
+      let firstNonSpaceOrTabIndex =
+        child.trailingTrivia.firstIndex(where: { !$0.isSpaceOrTab }) ?? child.trailingTrivia.endIndex
```
Comment (ahoppen) [inline CodeGeneration/Sources/Utils/CodeGenerationFormat.swift:163]: I think you can avoid the `??` if you push the optional unwrapping into the `if` condition.

Comment (roopekv) [reply CodeGeneration/Sources/Utils/CodeGenerationFormat.swift:163]: Like this?
```swift
if let firstNonSpaceOrTabIndex = child.trailingTrivia.firstIndex(where: { !$0.isSpaceOrTab }) {
  if child.trailingTrivia[firstNonSpaceOrTabIndex].isNewline {
    child.trailingTrivia = Trivia(pieces: child.trailingTrivia.suffix(from: firstNonSpaceOrTabIndex))
  }
} else {
  child.trailingTrivia = Trivia()
}
```

Comment (ahoppen) [reply CodeGeneration/Sources/Utils/CodeGenerationFormat.swift:163]: Yes, that’s what I was thinking

Comment (roopekv) [reply CodeGeneration/Sources/Utils/CodeGenerationFormat.swift]: The change has been made 👍

**Review by ahoppen — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by roopekv** [comment]
Could this be merged?

**Comment by ahoppen** [comment]
Thanks for the ping and sorry, I forgot about this PR. I’m not sure why CI failed and we don’t have the logs anymore. Let me re-trigger it and then we can see.

@swift-ci Please test

**Comment by ahoppen** [comment]
Looks like the Windows failure was because it picked up https://github.com/swiftlang/swift/pull/77149 but not https://github.com/swiftlang/swift-syntax/pull/2859. Let’s try again.

@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**